### PR TITLE
171648116 Add signup UI

### PR DESCRIPTION
**Pull request by Agnes-N**
#### Description

This PR is for creating SignUp UI. 

#### Type of change

Please select the relevant option

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Others (cosmetics, styling, improvements)
- [ ] This change requires a documentation update

#### How Has This Been Tested?

- [X] Unit
- [ ] Integration
- [ ] End-to-end
- [ ] No testing required

#### How to Test

> click the link in the conversation to view the deployment

#### Any background context you want to add (Optional)

N/A

#### STORY ID

[Finishes #171648116]

### CODE REVIEW

File: src/Components/Auth/SignUp/SignUp.jsx [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+    const { signUpAction } = this.props;
+    const { firstname, lastname, username, email, password } = this.state;
+    const form = {
+      firstname,
+      lastname,
+      username,
+      email,
+      password
+    };
+    await signUpAction(form);
```
Comment (brianineza01) [inline src/Components/Auth/SignUp/SignUp.jsx]: Kindly remove `async `and `await ` as you have already used them in actions

File: src/Components/Auth/SignUp/SignUp.jsx [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+    const { status } = this.props;
+    if (status === 201) {
+      this.infoToastify(
```
Comment (brianineza01) [inline src/Components/Auth/SignUp/SignUp.jsx]: Kindly use `toasts`, In `actions` where applicable

**Review by brianineza01 — CHANGES_REQUESTED** [review]

File: src/Components/Auth/SignUp/SignUp.jsx (hunk 2 above)
Comment (Agnes-N) [reply src/Components/Auth/SignUp/SignUp.jsx]: Done

File: src/Components/Auth/SignUp/SignUp.jsx (hunk 1 above)
Comment (Agnes-N) [reply src/Components/Auth/SignUp/SignUp.jsx]: Done

**Review by brianineza01 — DISMISSED** [review]

File: .eslintrc.json [hunk 3]
```diff
@@ -9,6 +9,9 @@
   },
   "extends": ["airbnb", "plugin:prettier/recommended", "prettier/react"],
   "rules": {
+    "indent": ["warn"],
+    "quotes": ["warn", "double"],
+    "no-tabs": "off",
```
Comment (placideirandora) [inline .eslintrc.json]: Turn it back on and refer to the documentation.

File: .eslintrc.json [hunk 4]
```diff
@@ -17,6 +20,11 @@
     "global-require": 0,
     "eslint linebreak-style": [0, "error", "windows"],
     "react/jsx-filename-extension": [1, { "extensions": [".js", ".jsx"]}],
+    "jsx-a11y/anchor-is-valid": [0],
+    "jsx-a11y/no-noninteractive-element-interactions": "off",
+    "jsx-a11y/click-events-have-key-events": "off",
+    "jsx-a11y/no-static-element-interactions": "off",
+    "jsx-a11y/control-has-associated-label": "off",
```
Comment (placideirandora) [inline .eslintrc.json]: Turn it back on and refer to the documentation.

File: __tests__/Components/Shared/AuthSidebar/AuthSidebar.test.js [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+      wrapper = setUp(props);
+    });
+
+    it("should test for a sign-up link", () => {
+      const component = setUp({
+        active: "signUp"
+      });
+      const SignUpLink = component.find("#sign-up-link");
+      expect(SignUpLink.props().className).toBe(
+        "auth-sidebar__nav__links auth-sidebar__nav__links--active"
```
Comment (placideirandora) [inline __tests__/Components/Shared/AuthSidebar/AuthSidebar.test.js]: Improve the BEM naming convention.

File: package.json [hunk 6]
```diff
@@ -55,12 +55,12 @@
...
     "sinon": "^9.0.2",
     "style-loader": "^1.2.1",
-    "styled-components": "^5.1.0",
     "svg-url-loader": "^5.0.0",
     "url-loader": "^4.1.0",
     "webpack": "^4.43.0",
     "webpack-cli": "^3.3.11",
-    "webpack-dev-server": "^3.11.0"
+    "webpack-dev-server": "^3.11.0",
+    "styled-components": "^5.1.0"
```
Comment (placideirandora) [inline package.json]: Kindly uninstall/remove this package.

File: package.json [hunk 7]
```diff
@@ -72,6 +72,9 @@
     "enzyme-adapter-react-16": "^1.15.2",
     "express": "^4.17.1",
     "jwt-decode": "^2.2.0",
+    "redux-mock-store": "^1.5.4",
+    "fetch-mock": "^9.10.0",
+    "node-fetch": "^2.6.0",
```
Comment (placideirandora) [inline package.json]: Kindly uninstall/remove this package.

File: src/Components/Auth/SignUp/SignUp.scss [hunk 8]
```diff
@@ -0,0 +1,217 @@
...
+  flex-direction: column;
+  height: 563px;
+  margin: 30px auto;
+  padding: 20px;
+  position: relative;
+  text-align: center;
+  top: 5%;
+  width: 90%;
+
+  &__h1 {
```
Comment (placideirandora) [inline src/Components/Auth/SignUp/SignUp.scss]: Change the class name. We should not use reserved keywords as our class names.

**Review by placideirandora — CHANGES_REQUESTED** [review]

**Review by placideirandora — APPROVED** [review]

**Pull request merged**